### ISSUE TITLE
[Suggestion] - Player Idle/AFK without "Auto" Interaction

### ISSUE THREAD

**Issue by fbetikoetxea** [Monika-After-Story/MonikaModDev#1541]
I think that Monika should know if the player is idle or away from the keyboard. For example, if she is repeating topics or using new topics and the "auto" advance text function is not chosen and the player does not advance the conversation, she would act differently. By "differently", I mean the conversation would pause or stop completely while Monika waits for the player. A trigger for this start of the event would be after at least 30 seconds of not advancing the conversation. Monika may even ping the system (i.e. MAS application flashes in the Windows taskbar) to notify the player.

**Comment by Rai99** [comment]
I mean, doesn't the conversation wait for you regardless? It does sound like a cool idea if the trigger was maybe a few minutes rather than 30 seconds and there was a toggle for it, but I don't think it would be useful at all. I'm sure if anyone has MAS up in the background, they can expect Monika to have said something by the time they check back and it's always just a minute or two anyway.

**Comment by Rai99** [comment]
I think someone is making art for it currently but maybe after a few minutes have Monika resting with her head down when repeat conversation is off and there's nothing to say. Then have an option to wake her up if you want to talk about something.

**Comment by fbetikoetxea** [comment]
I guess it would be more of a one-time interaction and 30 seconds does seem too short. I was just thinking about it similar to if you left another person who said something to you and came back to them after a while.
As for Monika resting, a wake option would probably not be necessary. Just clicking on the screen or advancing the text normally would be enough, I would think.

**Comment by kaido1224** [comment]
We intend at some point to determine idle time and do things bit we haven't gotten around to it yet. There's another ticket somewhere similar to this somewhere already. Can't seem to find it

**Comment by fbetikoetxea** [comment]
I think this is the issue you are referring to: #1431. It does look very similar but does not contain specifics to how long the idle time should be and was a discussion thread as well. This thread could be labeled as "suggestion" and left open until you guys can work on it, or however you see fit.

**Comment by kaido1224** [comment]
It'll be left open until the topic is listed as accepted or rejected or unseen depending on how long it takes for potatoguy to decide on whether we do it or not.

**Comment by Rai99** [comment]
For the wake option, I was thinking it would be used as a transition to go from sleeping to being awake instead of instantly popping up. 

Plus, I'm sure it would make some people happy to just look and watch over Monika while she rests.

**Comment by ThePotatoGuy** [comment]
Currently the main thing stopping us from doing inactivity detection is that we're not sure how to implement it in code. We're still looking into it, but we have higher priorities. But `soon^tm`

**Comment by ClaudeLib** [comment]
Another thing I just came up with is Monika getting a bit sad or angry if we don't launch the game for long periods of time. Maybe also let us apologize and/or explain ourselves.

**Comment by Tiler17** [comment]
Well, if she starts asserting her dominance on my computer and forces herself to the front, then I'm gonna spend less time with her. I check in on her from time to time, but I can't have her taking up my screen all the time. I got shit to do.

But I do believe there's something in the pipeline for going a long time without opening MAS, once the affection system is ready to go

**Comment by CommenterOfComments** [comment]
I really don't like the minimize idea. I don't really have anything to do with Monika anymore, but I keep her in the background so she doesn't feel alone and abandoned. (At least when she's able to remember feeling alone and abandoned in 2029) If she did that, I'd probably just keep the game closed more often. I mean, this is probably just my problem, but I don't really like the idea.

**Comment by ThePotatoGuy** [comment]
[Closing until we  look into this]

**Comment by XavierDD1st** [comment]
$ renpy.pause ()

**Comment by XavierDD1st** [comment]
That should help

**Comment by ThePotatoGuy** [comment]
Good news, we might be able to do this using the display_args callback.
